nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-actions.js

```diff
@@ -845,6 +845,116 @@
+                        var uri, entity;
+                        uri = config.urls.api + '/flow/process-groups/' + encodeURIComponent(d.id) + '/controller-services';
+                        entity = {
+                            'id': d.id,
+                            'state': 'ENABLED'
+                        };
+
+                        startRequests.push(updateResource(uri, entity).done(function (response) {
+                            nfCanvasUtils.getComponentByType('ProcessGroup').reload(d.id);
+                        }));
+                    });
+
+                    // inform Angular app once the updates have completed
+                    if (startRequests.length > 0) {
+                        $.when.apply(window, startRequests).always(function () {
+                            nfNgBridge.digest();
+                            if(typeof cb == 'function'){
+                                cb();
+                            }
+                        });
+                    } else if(typeof cb == 'function'){
+                         cb();
+                    }
+                }
+            }
+        },
+
+        /**
+         * Disable all controller services in the specified ProcessGroup.
+         *
+         * @argument {selection} selection      The selection
+         * @argument {cb} callback              The function to call when request is processed
+         */
+         disableAllControllerServices: function (selection,cb) {
+            if (selection.empty()) {
+                // build the entity
+                var entity = {
+                    'id': nfCanvasUtils.getGroupId(),
+                    'state': 'DISABLED'
+                };
+                updateResource(config.urls.api + '/flow/process-groups/' + encodeURIComponent(nfCanvasUtils.getGroupId()) + '/controller-services', entity).done(updateProcessGroup);
+            } else {
+                var componentsToStop = selection.filter(function (d) {
+                    return nfCanvasUtils.isProcessGroup(d3.select(this));
+                });
+
+                // ensure there are some component to stop
+                if (!componentsToStop.empty()) {
+                    var stopRequests = [];
+
+                    // stop each selected component
+                    componentsToStop.each(function (d) {
+                        var selected = d3.select(this);
+
+                        // prepare the request
+                        var uri, entity;
+                        uri = config.urls.api + '/flow/process-groups/' + encodeURIComponent(d.id) + '/controller-services';
+                        entity = {
+                            'id': d.id,
+                            'state': 'DISABLED'
+                        };
+
+                        stopRequests.push(updateResource(uri, entity).done(function (response) {
+                            nfCanvasUtils.getComponentByType('ProcessGroup').reload(d.id);
```

Comment:
Same comment as above regarding the `reload` call.

Reply:
Same comment above to keep this. :)